### ISSUE TITLE
Fields' label can't be set locally to inform assistive technology users what to input (1.3.1)

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux-datetime#157]
### Expected behavior

- Input field has a label programatically associated to be provided to assistive technology users about what to input. 
- Teams can associate the input to their context-specific label or provide a label.
- Default label doesn't include name of input component and is only "Date", "Time".

In example below, the input labels should be "Start time" and End "time"
![image](https://user-images.githubusercontent.com/15878597/82568155-686cc500-9b4c-11ea-8564-87fef94401d5.png)

### Actual behavior
Both the datepicker and timepicker have aria-label="Time input field" or "Date input field". Aria-label is at the top of the hierarchy in the algorithm browsers use to determine an input's label. Users will always get this as the label and not whatever the label needs to be given what's being asked for (e.g., Start time, End time).

### Impact
Assistive technology users won't be communicated correct input field label causing confusion and/or incorrect input.

**Comment by Blackbaud-TrevorBurch** [comment]
@Blackbaud-MattGregg overwriting the default works today (just verified this locally). However, I created a PR which updates the defaults to "Date" and "Time" instead of "Date input field" and "Time input field".

Just a quick reminder (just to be super clear) that when binding an aria label you must use [attr.aria-label] and not [aria-label].

**Comment by Blackbaud-MattGregg** [comment]
@Blackbaud-TrevorBurch Hi Trevor - noob question. In trying this out in StackBlitz - where does that [attr.aria-label] get set to bind? Does it go in the `<sky-datepicker>`'s     `<input`  element? I was only getting a parse error when I was trying this out.

**Comment by Blackbaud-TrevorBurch** [comment]
@Blackbaud-MattGregg Yup it goes on the input. Here is an example in StackBlitz https://stackblitz.com/edit/angular-bvkcku

**Comment by Blackbaud-MattGregg** [comment]
Thanks! I see what my issue was now - needed an extra set of single quotes on this type of attr I guess. I was missing those. Definitely see what you're saying on how teams can overwrite.

**Comment by blackbaud-johnly** [comment]
Do we need to add something to the docs to alert consumers about the need to overwrite and explain how?

**Comment by Blackbaud-TrevorBurch** [comment]
Hmm. That is a good question. @Blackbaud-MattGregg thoughts? We have the default in place to where we give it a label and the override is just native Angular and nothing we built in so should we show this in the code example or just let it be? I don't think we need explicit docs either way but the code example would be an option.

**Comment by Blackbaud-MattGregg** [comment]
I think providing good demos and examples that demonstrate using the components in a high accessible way should be the standard. When these get updated, they could be improved in this way to show how these should be used with both a visible label and the same available programmatically via aria-label.
I also don't think having an accessibility note would hurt to explain how to ensure its implemented accessibly (i.e., the same things - field should be used with a visual label and have the aria-label set correctly)

**Comment by Blackbaud-TrevorBurch** [comment]
In `4.0.1`

@blackbaud-johnly lets go ahead and make a docs issue for this and the phone field mirror change.
